packages/patterns/src/types.js

```diff
@@ -524,17 +524,17 @@ export {};
 /** @typedef {(...args: any[]) => any} Method */
 
 /**
- * @template {Record<string | symbol, MethodGuard>} [T=Record<string | symbol, MethodGuard>]
+ * @template {Record<PropertyKey, MethodGuard>} [T=Record<PropertyKey, MethodGuard>]
  * @typedef {{
  *   interfaceName: string,
- *   methodGuards: { [K in keyof T]: K extends symbol ? never : T[K] },
+ *   methodGuards: { [K in keyof T as Exclude<K, symbol>]: T[K] },
```

Comment:
This _looks_ better, but I don't think it actually works correctly: [TypeScript Playground](https://www.typescriptlang.org/play?target=7&filetype=js&useJavaScript=truee=4#code/PQKhCgAIUgBAXAngBwKYBNUDNIG8CuAdgNaED2A7oQL6QCyq8AFmegOL4CGATulCMHDhQEaHHioAtsgA2nCXgBKqAMZleAHgAK3Mmm5IA0qkQAaeoxbsuvAHy0A2gBUAvMrWade1AcTGzFsysHDzotgC6-OIoGNh4uFGQkACWhBLcWJwqqABynJKoAFyQAM7w3KkA5qaJkAVB1qElxbiQDoYphJDEJmQ4TuHFHagAHhKE6CWliJIARmQykAD8kISoAG4+kMVO7eGQ1DVi1LRanIgyZJzoAPo5Gz78gsJgUQhSsvKoSqrq6Nq6fRGEzmBgNEJ2Ryudx-AHeXz+UGWYI2MKRMQIGKYHC4BJiJKpdKZbJ5ArFMoVQjVWr1KwQyYtNodVLdXr9SCcKYAURGKhk+EwGkM5hKM3mMgiOz2ByOMBOkDOFyutx5fIFqCeQhE0TQeAAFMhzpdrsVFcbbvdNtwAJSQFy2BVG5V3B7cWgCcBqQhlDkqbLIeBm52WrYuSCGpXXO0OiPmgDcWrAOu+uANTpNjsjKt5-MwtvtmfNN1VudQ7sEXp9WX9gfT2bVmDt4br0ebWYTnrI3vgkFm1wAkmkfMTvmG8UlOkSsrl8kVIAByABiZDI86OE9pKKaLSgE73kmKevzDtwh13e6SDgAymKFgA6JicwdlTiEbKDSBH1uZGQlVDrpJqHAIDwGrVAAyDa4XStPU+3QQcp2ya0EzAiC62LHN1VggchwyadkKAA)
```suggestion
 *   methodGuards: { [K in keyof T]: K extends symbol ? never : T[K] },
```

Reply:
In the playground, you show the example of getting a desirable type error when passing a badly-constructed InterfaceGuard to a function.  However, introducing that behaviour leads to type errors when passing around well-formed InterfaceGuards if they have any symbol methods on them.

An [addendum to your playground](https://www.typescriptlang.org/play?target=7&filetype=js&useJavaScript=truee=4#code/PQKhCgAIUgBAXAngBwKYBNUDNIG8CuAdgNaED2A7oQL6QCyq8AFmegOL4CGATulCMHDhQEaHHioAtsgA2nCXgBKqAMZleAHgAK3Mmm5IA0qkQAaeoxbsuvAHy0A2gBUAvMrWade1AcTGzFsysHDzotgC6-OIoGNh4uFGQkACWhBLcWJwqqABynJKoAFyQAM7w3KkA5qaJkAVB1qElxbiQDoYphJDEJmQ4TuHFHagAHhKE6CWliJIARmQykAD8kISoAG4+kMVO7eGQ1DViSSUz8zIMDSG8JUvF7uroGgAUPYh9kE4AlJAAZNNzBbmS5Wa5hI4wai0LScRAyMicdAAfRyGx8-EEwjAUQQUlk8lQSlUj20un0RhMwMswRsYUcrgenjJPgpARBNNCERxSDQmBwuASx066Uy2TyBWKZQqhGqtXqoNpzTwbQ6qW6vX6kE4UwAoiMVDJ8JgNIZzKdATIIjs9gcIUkAed2Y0bndIIynq8NZ8fv9zecqVdabY7VDIDC4QjkXqDUbUBihCJomg8M9kLD4YjiuGM8jUZtuD8XLYw+nIyi0dxaAJwGpCGUtSpssh4Nmy3mti5IGmI4jIEWu6XEQBuBNgJOE3CpwfoLPTpHRw2YQvF1uI+f6xeoKuCWv1rJNltzhexvsDnvoPvF7s5kc1sh1+CQWaIgCSaR8osJncF9tSIqyuT5EUkAAOQAGJkGQIFHPa8ocjcLRQPayGSMUzzLnghxIchSQOAAymcCwAHRMNqb5lJwhDZIMkDoZekCZDIJSoDBSTUOA7HgPuqDNquuYVs8z7oG+-7ZF8I7cbxR4brGgmvu+GQAeJo4wAgMR4Hx5b5horSoas+BzD45j4YRMgkWRD6UdRxSEAZsxbNQ9jQDu971pUkHCQpn5aR2eDYcKH4AeKwHgZB0H+XBzqTIhOGwcUACMABMADM-lYchfoLE6YJKj+sUmRa5klOR8BWagNFJalyHsdQt6Jmpya4Jpx7GrpNl2UZbQEYVpHFZZVHle1hncAcTnVruj7uawImBdk64xpgp55QFilikBxShVBrF1NSUW5f5KEJSlaXbZlFy7TlMWxV1plFSVZUVcd1UcSOQA) shows how the `never` type prevents a legitimate value from being cast to an InterfaceGuard.

So that addendum shows the `never` typing raises the same error even for a well-formed InterfaceGuard.   I want to be able to carry along the methodGuards (all PropertyKeys) as a type parameter so that we can manipulate it in the exo makers (as I've done for the checking of guards against methods).

I think it's related to covariance/contravariance, but I'm not familiar with those terms enough to use them.  If you have a better solution you can demonstrate in the playground, please do let me know.

Reply:
Alright, I think I've got it: [TypeScript Playground](https://www.typescriptlang.org/play?target=7&ssl=10&ssc=6&pln=10&pc=115&filetype=js&useJavaScript=truee=4#code/PQKhCgAIUgBAXAngBwKYBNUDNIG8CuAdgNaED2A7oQL6QCyq8AFmegOL4CGATulCMHBDQEaHHioAtsgA2nCXgBKqAMZleAHgAK3Mmm5IA0qkQAaeoxbsuvAHy0A2gBUAvMrWade1AcTGzFsysHDzotgC6-OIoGNh4uFGQkACWhBLcWJwqqABynJKoAFyQAM7w3KkA5qaJkAVB1qElxbiQDoYphJAAogAeKjL4mBrEJmQ4TuYliJIARmQyEcVO7eGQtABkkFo88MmcMhqt7Z09veVZ8CNjE1Mz84vhxYSoAG4+67Y1YknTcwsMBohXglAD8LTaHVSZwuKiuo0Q40gk1K9wWS2Rq3W3xg1FoO0QMjInD40EEwjA0TQeAAFMhOITiehigSiSSAJSQFy2bYMtnoWgCcBqQhlSBZbLIeCsplcyD0xkkrk8hX8gDcQnAIipqDwMpJgsEIrFlTIrAAkmkfJlsnKEkkkql0jbcvkipAAOSm1ge74O+pWYHoZp4KAO8OSFrUP1JaNh35omSAwM2YMtePhtoAZUTADomJwSpaypxCNknng4+G49QNRLUFL9egad70JbnVlUOyNRSYAgYnq+UyjnVnvg5j5zA4c-8ZPnC8X4KXy2OJ9xPobhWRRfBIK329bOwB9Qg8XQUO0Zp2H7J5ArFL1m9Ans+UX0ZgPBVMh+2ZpKRyAAEYACYAGYY2xDM-geZMvyadM-ySac8wLIsd2XVAKxA8CMxrOsVElaUhxJFsnwPDJj1Pbhz27TVtX7alcCbTdjV3WYSXIl0j2ghZL3Da8KNvN0H3Y58eJkd8I0sOCQQQxCANwKtEOQ2d5zQksy0wqMILw8B60bYjm1Ezjj3E2jex1QdFXQEcAMIcdZknbMUIXdDNIrey1w3Mktx3SBjKtQTUCPSRkhKEoqj4h0BJdO93Q9USQrCiLCEqST-WkxpZMrCDxNgrK0xy8Baz0giGyI6yaQCjtsiS8KqnMrVKQY3UmMM2zV0c7gpxnB41MXDCPIcj5qHsHzWP8jjAq41Bzm4TgosdabOzikSSSPWaLnS-9MqDH8MwjYpsIg8NNvm7TcNyxN8r2uS-xUvrUIG9yjrAk6HQehZcyIEo1HpWYZFQEoK0UnSahK-SKv5Kqppq4Kzs4bsgA)

```
methodGuards: { [K in Exclude<keyof T, symbol>]: T[K] } & Partial<{ [K in Extract<keyof T, symbol>]: never }>
```

Reply:
> Alright, I think I've got it

Excellent!  Thank you for your tenacity. 😄

Is there anything else you'd like me to do as part of your Request Changes?